terraform/context_test.go

```diff
@@ -170,6 +171,27 @@ func resourceState(resourceType, resourceID string) *ResourceState {
 	}
 }
 
+// Test helper that gives a function 3 seconds to finish, assumes deadlock and
+// fails test if it does not.
+func testCheckDeadlock(t *testing.T, f func()) {
+	timeout := make(chan bool, 1)
+	done := make(chan bool, 1)
+	go func() {
+		time.Sleep(3 * time.Second)
+		timeout <- true
+	}()
+	go func(f func(), done chan bool) {
+		defer func() { done <- true }()
+		f()
+	}(f, done)
+	select {
+	case <-timeout:
```

Comment:
Actually, you can just remove all the above timeout stuff and replace this with `<-time.After(3 *time.Second)`